### DISCUSSION TITLE
periodic websocket failure with Cardano Ogmios running in docker container

### DISCUSSION THREAD

**Discussion by bensacc** [CardanoSolutions/ogmios#180] [Q & A] [answered]
I have a Cardano Ogmios image running in Docker on a VM, which will be running perfectly fine for some period of time (most recently about a week), and then suddenly start throwing this error:

{"severity":"Error","timestamp":"2022-02-14T16:50:26.801114665Z","thread":"11","message":{"WebSocket":{"tag":"WebSocketFailedToConnect","ioException":"Network.Socket.connect: <socket: 17>: does not exist (No such file or directory)"}},"version":"v5.1.0"}

Starting/Stopping the Docker container does not seem to resolve the issue, but deleting/recreating it does (with the unfortunate side effect of then taking several hours to resynch). Here is the docker command in its' entirety:

docker run -p 1337:1337 --name cardano-node-ogmios -d --restart=always -v cardano-node-ogmios-db:/ipc cardanosolutions/cardano-node-ogmios:latest

Not much to it, as you can see. I'm not sure if it's an Ogmios issue or a maybe a Docker issue? Has anyone experienced this before or have any thoughts?? I'm a bit stumped...thanks!

**Comment by KtorZ** [comment]
This error typically suggests that your cardano-node isn't up-and-running and therefore, Ogmios fails to establish connection with it. 

Any errors to report from your node itself? Docker containers are known to have limited max resources by default, RAM in particular, and mainnet has grown quite large recently so it is possible that it goes out of boundary here and dies. So please:

1. Make sure that your associated cardano-node process is still up and running
2. If the cardano-node isn't running, try to increase resources assigned to your container and see if that helps.

**Reply by bensacc** [reply]
thanks KtorZ, good thoughts!  In this case, I'm actually not running a separate cardano-node, but just Ogmios, which I was under the impression also created and synced a node for it's own use under localhost (I'm a bit new to this, though, so could certainly be wrong!).  It's definitely synching against something, as it's 26% of the way there after a restart.  Either way, I'm thinking I may have a memory issue so will bump that up  quite a bit.  thanks again for your help!

**Comment by KtorZ** [comment, marked as answer]
Given the command you've shown above I believe you are indeed running a node but likely don't realize it 😅
The `cardano-node-ogmios` docker image bundles as the name hints, a `cardano-node` and `ogmios` both in one container. The idea is to create one logical container which is a Cardano node with JSON/websocket support. Still, that's a full node behind!

I know that on MacOS, the default limit is 2GB of RAM allocated to containers which will definitely be not enough for mainnet. On Unix the limit is a bit higher as I recall but not sure about the exact number (nor if there's even a limit by default).

So you definitely wants to check that out and make sure that your system requirements are enough to run a full node.
